pkg/shell.go

```diff
@@ -20,3 +23,82 @@ func ExecShellReturnStrings(command string, args ...string) (string, string, err
 	log.Println("Commad Execution STDERR: %s", errb.String())
 	return outb.String(), errb.String(), err
 }
+
+func ExecShellReturnStringsWithVars(osvars map[string]string, command string, args ...string) (string, string, error) {
+	
+	log.Printf("INFO: Running %s",command)
+	for k, v := range osvars {		
+		os.Setenv(k, v)
+		log.Printf(" export %s = %s", k, v)
+	}
+	var outb, errb bytes.Buffer
+	k := exec.Command(command, args...)
+	k.Stdout = &outb
+	k.Stderr = &errb
+	err := k.Run()
+	if err != nil {
+		log.Println("Error executing command: %v", err)
+	}
+	log.Println("Commad Execution: %s", command)
+	log.Println("Commad Execution STDOUT: %s", outb.String())
+	log.Println("Commad Execution STDERR: %s", errb.String())
+	return outb.String(), errb.String(), err
+}
+
+// Exec shell actions supporting:
+// - On-the-fly logging of result
+// - Map of Vars loaded
+func ExecShellWithVars(osvars map[string]string, command string, args ...string) (error) {
+	
+	
+	log.Printf("INFO: Running %s",command)
+	for k, v := range osvars {		
+		os.Setenv(k, v)
+		log.Printf(" export %s = %s", k, v)
+	}
+	cmd := exec.Command(command, args...)
+	cmdReaderOut, _ := cmd.StdoutPipe()
```

Comment:
it would be nice to treat the errors

Reply:
I updated to return all errors to the caller. And log that something happen too.